src/EA.Weee.Web/Areas/Admin/Controllers/ProducerSubmissionController.cs

```diff
@@ -129,16 +136,89 @@ public async Task<ActionResult> AddPaymentDetails()
             return View();
         }
 
+        [AdminSmallProducerSubmissionContext]
         [HttpGet]
-        public async Task<ActionResult> RemoveSubmission()
+        public ActionResult RemoveSubmission(string registrationNumber, int year)
         {
-            return View();
+            var submission = this.SmallProducerSubmissionData.SubmissionHistory[year];
+            var selectedValue = string.Empty;
+            var model = new ConfirmRemovalViewModel
+            {
+                SelectedValue = selectedValue,
+                Producer = new ProducerDetailsScheme
+                {
+                    ComplianceYear = submission.ComplianceYear,
+                    ProducerName = submission.CompanyName,
```

Comment:
@stevelee-civica Thought I commented on this. Could be worth checking with James. In the case where the is authorised representative data it could be the the name should be the producer name rather than organisation name...

Reply:
waiting James to confirm it

Reply:
updated